### ISSUE TITLE
uskyblock not saving island worldguard regions/protection in uuid form

### ISSUE THREAD

**Issue by Wazez** [rlf/uSkyBlock#241]
hello, using latest uskyblock and its not saving the user as their uuid in worldguard regions. this causes players islands to get wiped when they change their name:

12000,-13250island:
min: {x: 11875.0, y: 0.0, z: -13375.0}
flags: {pvp: deny, greeting: Â§d** You are entering Â§bKrygon_43's Â§disland.,
farewell: Â§d** You are leaving Â§bKrygon_43's Â§disland., entry: allow}
max: {x: 12125.0, y: 255.0, z: -13125.0}
priority: 100
type: cuboid
owners:
players: [krygon_43, wazez]

when i use normal worldguard to add them, it stores their uuid. this means that the way uskyblock is creating and adding members is not doing it through uuid.

it should be:
12000,-13250island:
min: {x: 11875.0, y: 0.0, z: -13375.0}
flags: {pvp: deny, greeting: Â§d** You are entering Â§bKrygon_43's Â§disland.,
farewell: Â§d** You are leaving Â§bKrygon_43's Â§disland., entry: allow}
max: {x: 12125.0, y: 255.0, z: -13125.0}
priority: 100
type: cuboid
owners:
unique-ids: [7a01a16b-4434-4b8a-b6b4-7adf3e5d756a, 2b7d7438-2fd2-4181-b62c-5108f00b3cc8]

uskyblock version: v2.2-HF7
worldguard version: (latest) 1582. also tested on worldguard 6.0 and 1571

**Comment by rlf** [comment]
So, you still haven't learned, that "using latest skyblock" doesn't quite cut it?
Exactly what are you experiencing, apart from the region files looking as the above files.

**Comment by rlf** [comment]
You are correct, in noting that we use names instead of UUID when registering owners in WG-regions, but WG should be able to handle this for us.
And, it shouldn't pose a problem - since we handle name-changes on UUID extensively in uSkyBlock - including updating the WG-region manually.

**Comment by dutchy1001** [comment]
There is a setting in WG that convert all "old" names into UUID. You can find that setting in the main config of the WG folder. This needs to be done to have all island regions already on your server to be converted.

**Comment by Wazez** [comment]
"There is a setting in WG that convert all "old" names into UUID. You can find that setting in the main config of the WG folder. This needs to be done to have all island regions already on your server to be converted."

The worldguard conversion system is a one time thing. It does it on server startup and then disables. For this to work, we'd have to set it to true on "convert on startup" everytime

**Comment by rlf** [comment]
Let me try again:
So, you still haven't learned, that "using latest skyblock" doesn't quite cut it?
Exactly what are you experiencing, apart from the region files looking as the above files.

Our plugin (apart from any obvious/existing bugs) should take care of updating the WG region - so, what are you experiencing as a bug? Please provide information, instead of telling us what we do, and what you think we should do.

**Comment by Wazez** [comment]
it says:
uskyblock version: v2.2-HF7
worldguard version: (latest) 1582. also tested on worldguard 6.0 and 1571

when users create an island or join another island, uskyblock stores their name in the region.yml file under their region protection. instead it should be storing it as uuid.

i have tested this by adding the users manually with /rg addowner (region) (name) - it stores them in uuid. but if you do it with uskyblock it stores just the name

uskyblock should store the uuid of the player in the region.yml and not just the name.

**Comment by rlf** [comment]
You are still saying something about what we do, not what bug you are experiencing.

You are correct in noting what we do, but you are incorrect in assuming this is a problem.

Is this a problem for you? If so, please describe the problem!

**Comment by Wazez** [comment]
the problem is that when users change their name they lose their island because the region protection is not in uuid form

**Comment by rlf** [comment]
Huh, then report that as an error - hopefully with a step-by-step description of how it happens.
And preferably based on the v2.2-HF8 release (the latest - right now).

**Comment by Wazez** [comment]
is there a downloadable jar yet?

**Comment by rlf** [comment]
Try https://github.com/rlf/uSkyBlock/releases/tag/v2.2-HF8

**Comment by Wazez** [comment]
issue still remains.

using uskyblock2.2-HF8

i ran /is invite wazez
it stores the data as follows in regions.yml (worldguard)

    12000,-13250island:
        min: {x: 11875.0, y: 0.0, z: -13375.0}
        flags: {pvp: deny, farewell: Â§d** You are leaving Â§bKrygon_43's Â§disland.,
            greeting: Â§d** You are entering Â§bKrygon_43's Â§disland., entry: allow}
        max: {x: 12125.0, y: 255.0, z: -13125.0}
        priority: 100
        type: cuboid
        owners:
            players: [krygon_43, wazez]
        members: {}

if i remove wazez from the region with worldguard (/rg removeowner) and then add them with /rg addowner 12000,-13250island wazez
it will store the uuid of that player

    12000,-13250island:
        min: {x: 11875.0, y: 0.0, z: -13375.0}
        flags: {pvp: deny, farewell: Â§d** You are leaving Â§bKrygon_43's Â§disland.,
            greeting: Â§d** You are entering Â§bKrygon_43's Â§disland., entry: allow}
        max: {x: 12125.0, y: 255.0, z: -13125.0}
        priority: 100
        type: cuboid
        owners:
            players: [krygon_43]
            unique-ids: [f210610d-5f41-4d00-bf8a-61c2ed28f6cc]

its the way uskyblock is adding the user to the region.
if the user "wazez" changes their name, they wont be able to build or break blocks in that island.

**Comment by DatGamerGuy** [comment]
@rlf Woah. Stop being so rude. 
A key part of every single Developer is being able to accept and debug mistakes, and you are quite frankly treating @Wazez like a child over this. 

Also, you are wrong anyways, it's happening to me. 

Thank you and sorry for my apparent rudeness, however from the way you've acted in this thread, I think you need to change your attitude a bit, as the bug reporters are helping you.

**Comment by dutchy1001** [comment]
In the latest version (to keep in trent with the not knowing what is the latest) the regions should not have the UUID to it as USB is changing the names throughout the file when needed. When USB notest the a different name to the in USB stored UUID it automatic changes the WG region according to the name. This way WG doent need to be converted anymore and can work as it is. When server owners are "messing" with the files themself the chance is very big that there will be mistakes made. Adding owners to a skyblock region with the WG commands is in this case "messing up" the way USB works with WG. Incase you need to add a player manual to it keep it in line with how USB adds the player, in this case by name only. If you use WG in other worlds then you will need to the UUID as the owner.

Regarding to the rudeness @DatGamerGuy , when we try to answer as good and as quick we can it's not always possible as we have more then USB in our lives. As asked before and as guideline in the wiki or on the forums we ask not to use the term "the latest" for USB or any plugin we have as dependend. If we then get pm's or same questions in different forums you can imagine that it sometimes is hard to stay polite.

As I did what a lot of ppl do in the beginning of my post to state " the latest" on purpose and a bit sarcastic I will tell what versions I used:
* git-Spigot-d0d1d87-15e81cf (MC: 1.8)
* uSkyblock v2.2-HF8
* WorldEdit v6.0;3342-78f975b9
* WorldGuard v6.0.0-beta-05.1569
* Vault v1.5.2-b33
* Essentials vTeamCity
* GroupManager v2.1.31 (Phoenix)
* Multiverse-Core v2.5-b691

**Comment by rlf** [comment]
@DatGamerGuy I'm usually not rude - but Wazez has repeatedly tested my patience - posting both on forums, here and contacting me by PM. He has repeatedly ignored our (both @dutchy1001 and my) attempt to get him to clarify things - and I AM a grumpy old man - once you get on my bad side, you stay there. You can rest assured though - that I will not treat you with the same contempt :D

**Comment by DatGamerGuy** [comment]
@rlf @dutchy1001 Ah ok guys. Sorry about the mixup - I was very shocked by what I saw here.

However, this problem is still happening to me yet I have tried your solutions.

**Comment by rlf** [comment]
What problem? That the worldguard region file is storing owners with names instead of UUIDs? Or something more tangible?
Because, as mentioned earlier - yes - it stores the names as owners - but it's not a problem (it uses names a lot of places) - but it (in turn) handles the name-change when it occurs.

**Comment by DatGamerGuy** [comment]
@rlf It does store as names yes, however if I add a region name manually through /rg, it'll update when the name is changed, but if USB does it, it will not update when the name is changed.

**Comment by VoidWhisperer** [comment]
I think the issue here is this: It's storing it as a name, and as you say WG should handle that automatically. However, it isn't, as people are losing their islands when they change their name, hinting that the conversion isn't taking place.

Versions:
uSkyBlock version 2.3.0-SNAPSHOT
WG 6.0.0-SNAPSHOT.1573-

If there is anymore info I can include just ask.

**Comment by dutchy1001** [comment]
@DatGamerGuy  as I already said before adding with the WG command an owner is a bad idea, however if you do it in the config file itself with adding the name it should not cause a problem. The problem occurs when using the WG command. 

@VoidWhisperer I think the version you have was still having the issue's that the UUID wasnt working properly, update to v2.2HF9b and try again.

As a warning to this version, the purge does work but isnt fully tested on it reability. The use of the purge is at the server owners own risk. YOU BEEN WARNED
---

**Comment by rlf** [comment]
@VoidWhisperer as @dutchy1001 said - it sounds like you have a version that doesn't have the latest UUID fixes - the v2.3-SNAPSHOT isn't the latest release, even though it has the highest number - I'll remove it from the releases section.

**Comment by dutchy1001** [comment]
This can be closed as we know now how it works and there is no need for UUID in the "skyblock" regions.
